File: src/test/java/io/github/seggan/jyxal/runtime/CompressionTest.java
package io.github.seggan.jyxal.runtime;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class CompressionTest {

    @Test
    public void testCompress() {
        Assertions.assertEquals("Hello, World!", Compression.decompress("\u0188\u1E61, \u019B\u20AC!"));
        Assertions.assertEquals("This is a test String to decompress.", Compression.decompress("\u03BB\u00AB is a \u2228\u1E0A \u00F8\u1E8F to de\u2022\u215B\u27D1\u013F."));
        Assertions.assertEquals("A test string.", Compression.decompress("A test string."));
    }
}

File: src/test/kotlin/io/github/seggan/jyxal/runtime/CompressionTest.kt
package io.github.seggan.jyxal.runtime

import io.github.seggan.jyxal.runtime.text.Compression.compress
import io.github.seggan.jyxal.runtime.text.Compression.decompress
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class CompressionTest {
    @Test
    fun testCompress() {
        Assertions.assertEquals("Hello, World!", decompress("ƈṡ, ƛ€!"))
        Assertions.assertEquals(
            "This is a test String to decompress.",
            decompress("λ« is a ∨Ḋ øẏ to de•⅛⟑Ŀ.")
        )
        Assertions.assertEquals("A test string.", decompress("A test string."))

        Assertions.assertEquals("`ƈṡ, ƛ€!`", compress("Hello, World!"))
        Assertions.assertEquals(
            "`λ« is a ∨Ḋ øẏ to de•⅛⟑Ŀ.`",
            compress("This is a test String to decompress.")
        )
    }
}